src/Bicep.Core/TypeSystem/Radius/V3/CommonBindings.cs

```diff
@@ -149,15 +149,35 @@ public static class CommonBindings
             Type = new ThreePartType("redislabs.com", "Redis", RadiusResources.CategoryBinding),
             Properties =
             {
-                new TypeProperty("host", LanguageConstants.String, TypePropertyFlags.ReadOnly),
+                new TypeProperty("host", LanguageConstants.String, TypePropertyFlags.None),
                 new TypeProperty("port", LanguageConstants.Int, TypePropertyFlags.None),
                 new TypeProperty("username", LanguageConstants.String, TypePropertyFlags.ReadOnly),
+                // The secrets section allows usage of binding expression to specify
+                // custom secrets.
+                //
+                // TODO: It is slightly confusing to set `redis.secrets.connectionString`, and read
+                //       from `redis.connectionString`. However, it is backward compatible with the
+                //       existing model. This will be a point we need to discuss when
+                //       check-pointing the end-to-end for RedisComponent.
```

Comment:
They are two different scenarios. Currently we use `"passwords"` since the connectionString format isn't yet the right way.

My guest is that when we come out of the tunnel we won't need `"password"` anymore since `connectionString` will be fully customizable. Let me add another `TODO` here to raise this during discussion.

Reply:
Added the TODO